### PR TITLE
Improve on top of the work made by Cyril on channel API

### PR DESCRIPTION
**Pull request by G10h4ck**
Fix compilation, retrocompatibility and enums
    
    Workaround miss-behaviour on old Android phones
    Cleanup indentation a bit
    Consistent param naming
    Introduce default parameter values also for output paramethers

### CODE REVIEW

File: libretroshare/src/retroshare/rsgxsifacehelper.h [hunk 1]
```diff
@@ -292,25 +290,55 @@ struct RsGxsIfaceHelper
...
+		auto wkStartime = std::chrono::steady_clock::now();
+		int maxWorkAroundCnt = 10;
+LLwaitTokenBeginLabel:
+#endif
 		auto timeout = std::chrono::steady_clock::now() + maxWait;
 		auto st = requestStatus(token);
 		while( !(st == RsTokenService::FAILED || st >= RsTokenService::COMPLETE)
 		       && std::chrono::steady_clock::now() < timeout )
 		{
-			std::this_thread::sleep_for(std::chrono::milliseconds(2));
```
Comment (csoler) [inline libretroshare/src/retroshare/rsgxsifacehelper.h:305]: Sleeping for only 2 ms between two checks is probably far too small. You'll be creating lots of CPU loops. I would advise 100 or 200 ms here.

File: libretroshare/src/util/rsmemory.h [hunk 2]
```diff
@@ -23,8 +23,11 @@
 
 #include <stdlib.h>
 #include <iostream>
+#include <memory>
 #include <util/stacktrace.h>
 
+#define RS_DEFAULT_STORAGE_PARAM(Type) *std::unique_ptr<Type>(new Type)
```
Comment (csoler) [inline libretroshare/src/util/rsmemory.h:63]: Could you add a quick comment about this?

Comment (csoler) [reply libretroshare/src/util/rsmemory.h:63]: Congrats for the macro btw ;-)

**Comment by csoler** [comment]
I added a few comments inline. Everything seems clean.
I still think we should rename the old methods *_deprecated() and use the normal names for the new ones. This will force current developers to replace the names if they want to keep using the old format, but it's very easy to do, and it will guarranty that in 2-3 years when we remove the *_deprecated() methods and we hopefully have lot more developers using them, it wont annoy anyone and ensure that the name is the most appropriate.

**Comment by G10h4ck** [comment]
> I added a few comments inline. Everything seems clean.
> I still think we should rename the old methods *_deprecated() and use the normal names for the new ones. This will force current developers to replace the names if they want to keep using the old format, but it's very easy to do, and it will guarranty that in 2-3 years when we remove the *_deprecated() methods and we hopefully have lot more developers using them, it wont annoy anyone and ensure that the name is the most appropriate.

The problem with that is that for JSON API users the things that used to works, would break without much notice, and because we released a stable version with that API I would avoid breaking the API at least for a couple versions, otherwise JSON API will be not attractive at all other projects/developers :-\

**Comment by csoler** [comment]
Come on! It's not compiled by default so people who use it compile RS themselves. Changing the name is not a terrible task. It's anyway better than keeping this V2 name for years.

File: libretroshare/src/retroshare/rsgxsifacehelper.h (hunk 1 above)
Comment (G10h4ck) [reply libretroshare/src/retroshare/rsgxsifacehelper.h:305]: 100ms sound reasonable for operations like channel creation that take lot of time but it is far too much for simple reads, I have been measuring this value even on old android phones and for reads operations 2ms is more then enough (probably the system does some caching), also this PR doesn't change the default value it just make it configurable, I agree that long operations like channel creation might be optimized by passing a longer value as parameters ;)

**Comment by G10h4ck** [comment]
> Come on! It's not compiled by default so people who use it compile RS themselves. Changing the name is not a terrible task. It's anyway better than keeping this V2 name for years.

It is enabled at compile time in most linux distributions packages, and it is mandatory for retroshare-service ;)

**Pull request merged**